airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+        return self.body
+
+    @staticmethod
+    def _convert_date(field_date):
+        return {
+            'day': field_date.day,
+            'month': field_date.month,
+            'year': field_date.year
+        }
+
+    @staticmethod
+    def _convert_time(time):
+        return {
+            "hours": time.hour,
+            "minutes": time.minute,
+            "seconds": time.second,
+        }
+
+
+class TransferJobValidator:
+
+    def __init__(self, body):
+        self.body = body
+
+    def _verify_data_source(self):
+        if 'transferSpec' not in self.body:
+            return
+
+        is_gcs = 'gcsDataSource' in self.body['transferSpec']
+        is_aws_s3 = 'awsS3DataSource' in self.body['transferSpec']
+        is_http = 'httpDataSource' in self.body['transferSpec']
+
+        if not is_gcs ^ is_aws_s3 ^ is_http:
+            raise AirflowException("You must choose only one data source")
+
+    def _restrict_aws_credentials(self):
+        if 'transferSpec' in self.body and \
+           'awsS3DataSource' in self.body['transferSpec'] and \
+           'awsAccessKey' in self.body['transferSpec']['awsS3DataSource']:
+            raise AirflowException("Credentials in body is not allowed. "
+                                   "To store credentials, use connections.")
+
+    def _restrict_empty_body(self):
+        if not self.body:
+            raise AirflowException("The required parameter 'body' "
+                                   "is empty or None")
+
+    def validate_body(self):
+        self._restrict_empty_body()
+        self._restrict_aws_credentials()
+        self._verify_data_source()
+
+
+class GcpTransferServiceJobCreateOperator(BaseOperator):
+    """
+    Creates a transfer job that runs periodically.
+
+    .. warning::
+
+        This operator is NOT idempotent. If you run it many times, many transfer
+        job will be created on the Google Cloud Platform.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GcpTransferServiceJobCreateOperator`
+
+    :param body: The request body, as described in
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/create#request-body
+        With three additional improvements:
+
+        * dates can be given in the form datetime.date
+        * times can be given in the form datetime.time
+        * credentials to AWS should be stored in connection and
+          selected by the aws_conn_id parameter
+
+    :type body: dict
+    :param aws_conn_id: The connection ID used to retrieve credentials to
+        Amazon Web Service.
+    :type aws_conn_id: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud
+        Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    :return: Google Storage Transfer service object
+    :rtype: dict
```

Comment:
I think suggestion to describe it in plain English is good idea. But I am just wondering - why not describe it where it belongs - i.e. in execute() method's docstring?

Reply:
The problem is that the user, when creating a DAG, will not look at `execute()`'s docstring, because he never invokes it explicitly - it's Airflow underneath that does. Obviously if he looks into the operator's source code he'll see it, but if he only uses IDE's documentation quick lookup (for me it's `ctrl`+`j`) he'll only see the docstring of the class.